src/IOBuffer.ts

```diff
@@ -268,6 +278,74 @@ export class IOBuffer {
     return bytes;
   }
 
+  /**
+   * Create array of size and TypedArray type
+   * @param size - Size of array
+   * @param word - TypedArray type
+   * @returns
+   */
+  private makeTypedArray(word: Word, size = 1) {
+    switch (word) {
+      case 'int8':
+        return new Int8Array(size);
+      case 'uint8':
+        return new Uint8Array(size);
+      case 'int16':
+        return new Int16Array(size);
+      case 'uint16':
+        return new Uint16Array(size);
+      case 'int32':
+        return new Int32Array(size);
+      case 'uint32':
+        return new Uint32Array(size);
+      case 'float32':
+        return new Float32Array(size);
+      case 'float64':
+        return new Float64Array(size);
+      default:
+        throw new Error('Invalid word');
+    }
+  }
+  /**
+   * Select function for reading multiple values
+   * @param word - number type
+   * @returns - reader function to use elsewhere
+   */
+  private useReader(word: Word) {
+    switch (word) {
+      case 'int8':
+        return this.readInt8.bind(this);
+      case 'uint8':
+        return this.readUint8.bind(this);
+      case 'int16':
+        return this.readInt16.bind(this);
+      case 'uint16':
+        return this.readUint16.bind(this);
+      case 'int32':
+        return this.readInt32.bind(this);
+      case 'uint32':
+        return this.readUint32.bind(this);
+      case 'float32':
+        return this.readFloat32.bind(this);
+      case 'float64':
+        return this.readFloat64.bind(this);
+      default:
+        throw new Error('Invalid word type');
+    }
+  }
+  /**
+   * Makes typed array after reading bytes
+   * @param size - number of elements to read
+   * @param word - type of elements to read
+   */
+  public arrayOf(size: number, word: Word) {
+    const createArray = this.makeTypedArray(word, size);
+    const useReader = this.useReader(word);
+    for (let i = 0; i < size; i++) {
+      createArray.set([useReader()], i);
+    }
```

Comment:
This is very inefficient. I'd do it like this to avoid the loop and unnecessary bind and temporary arrays:

```ts
/*
  Assumes there is a map of TypedArray constructors.
  You can also use it above instead of the big switch block.
*/
type TypedArrayConstructor = {
  new(buffer: ArrayBufferLike, byteOffset: number, length: number): ArrayBufferView;
  BYTES_PER_ELEMENT: number;
}
const typedArrays: Record<Word, TypedArrayConstructor> = {
  uint8: Uint8Array,
  uint16: Uint16Array,
  // other constructors ...
};

const typedArray = new typedArrays[word](this.buffer, size * typedArrays[word].BYTES_PER_ELEMENT, this.offset);
createArray.set(typedArray, 0);

Reply:
Still refactoring but if I get your idea this is the only code remaining:
```
  /**
   * Makes typed array after reading bytes
   * @param size - number of elements to read
   * @param type - number type of elements to read
   */

  public readArray(size: number, type: NumberType) {
    const typedArray = new typedArrays[type](
      this.buffer,
      size * typedArrays[type].BYTES_PER_ELEMENT,
      this.offset,
    );
    this.offset += size * typedArrays[type].BYTES_PER_ELEMENT;
    return typedArray;
  }
```

Reply:
I'd still prefer to return a copy (hence the `createArray.set(typedArray, 0)` in my suggestion), to avoid keeping a reference on the underlying array buffer (which could be huge)

Reply:
I think I had some issue creating the copy. But I'll try. Also I think the second and third argument have to be swapped 
```
      size * typedArrays[type].BYTES_PER_ELEMENT,
      this.offset,
```

Reply:
Hopefully this is a bit better. This is more or less what I have now:
```typescript
  public readArray(size: number, type: keyof typeof typedArrays) {
    const bytes = typedArrays[type].BYTES_PER_ELEMENT * size;
    const bytesCopy = new Uint8Array(bytes);
    if (type === 'uint8') {
      return this.readBytes(bytes);
    }
    //offset needs to be 0 for the new data view (for example multiple for 4 for float32)
    bytesCopy.set(this.readBytes(bytes));
    
    return  new typedArrays[type](bytesCopy);  }
```